### PR TITLE
Kiknos NSE NAT support

### PR DESCRIPTION
**Pull request by rastislavs**
Source NAT on IPSec tunnel interfaces can be enabled with the helm option `ucnf.natIP`. Its value should contain the requested outside NAT IP address, e.g. `ucnf.natIP=1.2.3.4`.

Enabling NAT results into the following configuration on the NSE VPP:

- each NSC-facing interface is configured as "NAT inside"
- each IPSec (IPIP) tunnel interface is configured as "NAT outside"

```
vpp# sh nat44 interfaces                                                                                                                                                                                      
NAT44 interfaces:                                                                                                                                                                                             
 memif1/0 in                                                                                                                                                                                                  
 memif2/0 in                                                                                                                                                                                                  
 memif3/0 in                                                                                                                                                                                                  
 ipip0 out   
```

NAT pool is configured to the NAT IP provided in the `ucnf.natIP` helm option:

```
vpp# sh nat44 addresses 
NAT44 pool addresses:
1.2.3.4
  tenant VRF: 0
  0 busy other ports
  0 busy udp ports
  0 busy tcp ports                                                                                                                                                                                            
  0 busy icmp ports 
```

At this point, every conection from inside (NSCs) to outside (remote IPSec peers) is source-NATed to the configured NAT IP. Connections from outside to inside are not allowed (not to NAT IP nor NSC IPs). To allow outside-to-inside connections, NSC has to request forwarding of a specific TCP/UDP port to them. They can do it with lables in `networkservicemesh.io` annotations, e.g. 
`ns.networkservicemesh.io: nsm-service-name?nat-port-forward-tcp=80` to forward TCP port 80, or `ns.networkservicemesh.io: nsm-service-name?nat-port-forward-udp=53`
to port forward UDP port 53.


That renders into a static nat44 mapping configuration on VPP, e.g.:

```
vpp# sh nat44 static mappings                                                                                                                                                                                                               
NAT44 static mappings:                                                                                                                                                                                                                      
 tcp local 172.31.22.9:80 external 1.2.3.4:80 vrf 0  out2in-only                                                                                                                                                                            
```

In our setup, the Istio NSC requests port forwarding on port 80 with `ns.networkservicemesh.io: {{ .Values.nsm.serviceName }}?nat-port-forward-tcp=80`, which means that remote IPSec peers can access the Istio ingress gateway running in the cluster at `http://1.2.3.4:80`.

### CODE REVIEW

File: examples/ucnf-kiknos/scripts/start_topo.sh [hunk 1]
```diff
@@ -3,8 +3,8 @@
 # Topology information
 CLUSTER1=${CLUSTER1:-kiknos-demo-1}
 CLUSTER2=${CLUSTER2:-kiknos-demo-2}
-VPP_AGENT=${VPP_AGENT:-ciscolabs/kiknos:latest}
-NSE_ORG=${NSE_ORG:-mmatache}
+VPP_AGENT=${VPP_AGENT:-ciscolabs/kiknos-sswan:latest}
```
Comment (tiswanso) [inline examples/ucnf-kiknos/scripts/start_topo.sh:6]: did you need to update the vppagent for any NAT feature?

File: examples/ucnf-kiknos/helm/kiknos_vpn_endpoint/templates/kiknos-nse-ucnf.yaml [hunk 2]
```diff
@@ -188,12 +192,18 @@ metadata:
   name: ucnf-kiknos-{{ .Values.nsm.serviceName }}
 data:
   config.yaml: |
+    {{- if .Values.ucnf.natIP }}
+    initactions:
+    - dpconfig:
+        nat44pools:
```
Comment (tiswanso) [inline examples/ucnf-kiknos/helm/kiknos_vpn_endpoint/templates/kiknos-nse-ucnf.yaml:198]: Does this key map to: https://github.com/ligato/vpp-agent/blob/master/proto/ligato/vpp/nat/nat.proto#L126-L136

I guess the new build of the vppagent image creates the external nat44Interface config.

File: examples/ucnf-kiknos/scripts/start_topo.sh (hunk 1 above)
Comment (rastislavs) [reply examples/ucnf-kiknos/scripts/start_topo.sh:6]: I had to update kiknos agent (https://wwwin-github.cisco.com/ccnf/kiknos/pull/37), but not ligato vpp-agent.
Also, `kiknos-sswan` is what we call kiknos v2, refactored kiknos control plane.

File: examples/ucnf-kiknos/helm/kiknos_vpn_endpoint/templates/kiknos-nse-ucnf.yaml (hunk 2 above)
Comment (rastislavs) [reply examples/ucnf-kiknos/helm/kiknos_vpn_endpoint/templates/kiknos-nse-ucnf.yaml:198]: yes, it maps to the `Nat44AddressPool` proto that you referenced

**Pull request closed without merging**